explorer/src/logging.ts

```diff
@@ -1,46 +1,65 @@
-import {
-  requestWhitelist,
-  logger as loggerConfig,
-  errorLogger,
-} from 'express-winston'
-import * as winston from 'winston'
+import PinoHttp from 'express-pino-logger'
+import pino from 'pino'
 import express from 'express'
+import { Logger } from 'typeorm'
 
-const LOGGER_WHITELIST = [
-  'url',
-  'method',
-  'httpVersion',
-  'originalUrl',
-  'query',
-]
-requestWhitelist.splice(0, requestWhitelist.length, ...LOGGER_WHITELIST)
+const options: any = {
+  name: 'Explorer',
+  level: 'warn',
+  redact: {
+    paths: ['req.headers', 'res.headers'],
+  },
+}
+if (process.env.EXPLORER_DEV) {
+  options['prettyPrint'] = { colorize: true }
+  options['level'] = 'debug'
+} else if (process.env.NODE_ENV === 'test') {
+  options['level'] = 'silent'
+}
+export const logger = pino(options)
 
 export const addRequestLogging = (app: express.Express) => {
-  const consoleTransport = new winston.transports.Console()
-
-  app.use(
-    loggerConfig({
-      expressFormat: true,
-      meta: true,
-      msg: 'HTTP {{req.method}} {{req.url}}',
-      transports: [consoleTransport],
-    }),
-  )
-
-  app.use(
-    errorLogger({
-      transports: [consoleTransport],
-    }),
-  )
+  app.use(PinoHttp({ logger }))
 }
 
-const transports = {
-  console: new winston.transports.Console({
-    level: 'info',
-    silent: process.env.NODE_ENV === 'test',
-  }),
-}
+export class TypeOrmLogger implements Logger {
+  public logQuery(
+    query: string,
+    parameters?: any[],
+    queryRunner?: QueryRunner): any {
+    logger.trace({msg: query, parameters})
+  }
+
+  public logQueryError(
+    error: string,
+    query: string,
+    parameters?: any[],
+    queryRunner?: QueryRunner): any {
+    logger.error({msg: 'DB query failed', error, query, parameters})
+  }
 
-export const logger = winston.createLogger({
-  transports: [transports.console],
-})
+  public logQuerySlow(
+    time: number,
+    query: string,
+    parameters?: any[],
+    queryRunner?: QueryRunner): any {
+    logger.warn({msg: 'Slow DB query detected', duration: time, query, parameters})
+  }
+
+  public logSchemaBuild(
+    message: string,
+    queryRunner?: QueryRunner): any {
+    logger.trace({msg: message})
+  }
+
+  public logMigration(message: string, queryRunner?: QueryRunner): any {
+    logger.info({msg: message})
+  }
+
+  public log(
+    level: 'log' | 'info' | 'warn',
+    message: any,
```

Comment:
I would restrict this to `string | number | null | undefined` or something like that. Passing an object to a string templates results in it being cased to `[object Object]'

Reply:
The original interface is `any` so changing this would have no effect (not sure if it'd be accepted either).